### PR TITLE
GULUserDefaults clear for all archs

### PR DESCRIPTION
**Pull request by paulb777**

### CODE REVIEW

File: GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m [hunk 1]
```diff
@@ -424,15 +399,12 @@ - (void)testUserDefaultNotifications {
   [newUserDefaults setObject:nil forKey:@"test-another"];
   XCTAssertNil([newUserDefaults objectForKey:@"test-another"]);
   [newUserDefaults synchronize];
-  [newUserDefaults clearAllData];
   [[NSNotificationCenter defaultCenter] removeObserver:observer];
-
-  // Remove the underlying reference file.
```
Comment (ryanwilson) [inline GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m]: Did you mean to remove this comment?

File: GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m [hunk 2]
```diff
@@ -387,7 +362,7 @@ - (void)testNewSharedUserDefaultsWithStandardUserDefaults {
 
   // Remove all of the objects from the new user defaults. The values from the NSUserDefaults must
   // also be cleared.
-  [newUserDefaults clearAllData];
+  [[NSUserDefaults standardUserDefaults] removePersistentDomainForName:appDomain];
```
Comment (maksymmalyhin) [inline GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m]: Do we still need 365-379?

File: GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m (hunk 1 above)
Comment (paulb777) [reply GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m]: Nope. Good catch!

File: GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m (hunk 2 above)
Comment (paulb777) [reply GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m]: Nope. Deleted. Thanks!

Comment (maksymmalyhin) [reply GoogleUtilities/Example/Tests/UserDefaults/GULUserDefaultsTests.m]: No problem!

**Pull request merged**